crates/graphql/src/schema.rs

```diff
@@ -264,6 +600,47 @@ impl juniper::Context for AppContext {}
 
 #[juniper::graphql_object(Context = AppContext)]
 impl QueryRoot {
+    async fn profile(
+        &self,
+        ctx: &AppContext,
+        #[graphql(description = "Twitter handle")] handle: String,
+    ) -> Option<Profile> {
+        let twitter_bearer_token = &ctx.twitter_bearer_token;
+        let http_client = HttpClient::new();
+
+        let twitter_show_response = http_client
+            .get("https://api.twitter.com/1.1/users/show.json")
+            .header("Accept", "application/json")
+            .query(&[("screen_name", handle.clone())])
+            .bearer_auth(twitter_bearer_token)
+            .send()
+            .await
+            .ok()?
+            .json::<TwitterShowResponse>()
+            .await
+            .ok()?;
+
+        let twitter_profile_picture_response = http_client
+            .get(format!(
+                "https://api.twitter.com/2/users/by/username/{}",
+                handle.clone()
+            ))
+            .header("Accept", "application/json")
+            .query(&[("user.fields", "profile_image_url")])
+            .bearer_auth(twitter_bearer_token)
+            .send()
+            .await
+            .ok()?
+            .json::<TwitterProfilePictureResponse>()
+            .await
+            .ok()?;
```

Comment:
Is rust handling making these calls in parallel or does it require something like a promise all?